app/interactors/file_attachments/create_interactor.rb

```diff
@@ -34,11 +34,14 @@ def find_and_lock_edition
   end
 
   def upload_attachment
-    context.attachment_revision = FileAttachmentUploadService.new(
-      params[:file],
-      edition.revision,
+    blob_attributes = FileAttachmentBlobAttributesService.new(file: params[:file],
+                                                              revision: edition.revision)
+                                                         .call
+    context.attachment_revision = FileAttachment::Revision.create_initial(
```

Comment:
It might be a little clearer to have named params here, since the first two are a bit of a mix.

Reply:
Sounds good - it's inconsistent with the other create_initials but then the interfaces aren't the same so I think that's fine.